op-e2e/actions/ecotone_fork_test.go

```diff
@@ -0,0 +1,139 @@
+	require.True(t, scalar.Cmp(new(big.Int).SetUint64(dp.DeployConfig.GasPriceOracleScalar)) == 0, "must match deploy config")
+	t.Log("original scalar", scalar)
+
+	// Get current implementations addresses (by slot) for L1Block + GasPriceOracle
+	initialGasPriceOracleAddress, err := engine.EthClient().StorageAt(context.Background(), predeploys.GasPriceOracleAddr, genesis.ImplementationSlot, nil)
+	require.NoError(t, err)
+	initialL1BlockAddress, err := engine.EthClient().StorageAt(context.Background(), predeploys.L1BlockAddr, genesis.ImplementationSlot, nil)
+	require.NoError(t, err)
+
+	// Build to the ecotone block
+	sequencer.ActBuildL2ToEcotone(t)
+	block := sequencer.L2Unsafe()
+
+	// get latest block
+	latestBlock, err := engine.EthClient().BlockByNumber(context.TODO(), nil)
+	require.NoError(t, err)
+	require.Equal(t, block.Number, latestBlock.Number().Uint64())
+
+	transactions := latestBlock.Transactions()
+
+	l1Info, err := derive.L1BlockInfoFromBytes(sd.RollupCfg, latestBlock.Time(), latestBlock.Transactions()[0].Data())
+	require.NoError(t, err)
+	t.Log("seq num", l1Info.SequenceNumber)
+
+	// L1Block: 1 set-L1-info + 2 deploys + 2 upgradeTo + 1 enable ecotone on GPO + 1 4788 deploy
+	// See [derive.EcotoneNetworkUpgradeTransactions]
+	require.Equal(t, 7, len(transactions))
+
+	// All transactions are successful
+	for i := 1; i < 7; i++ {
+		txn := transactions[i]
+		receipt, err := engine.EthClient().TransactionReceipt(context.Background(), txn.Hash())
+		require.NoError(t, err)
+		require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+	}
+
+	expectedL1BlockAddress := crypto.CreateAddress(derive.L1BlockDeployerAddress, 0)
+	expectedGasPriceOracleAddress := crypto.CreateAddress(derive.GasPriceOracleDeployerAddress, 0)
+
+	// Gas Price Oracle Proxy is updated
+	updatedGasPriceOracleAddress, err := engine.EthClient().StorageAt(context.Background(), predeploys.GasPriceOracleAddr, genesis.ImplementationSlot, latestBlock.Number())
+	require.NoError(t, err)
+	assert.Equal(t, expectedGasPriceOracleAddress, common.BytesToAddress(updatedGasPriceOracleAddress))
+	assert.NotEqualf(t, initialGasPriceOracleAddress, updatedGasPriceOracleAddress, "Gas Price Oracle Proxy address should have changed")
+
+	// L1Block Proxy is updated
+	updatedL1BlockAddress, err := engine.EthClient().StorageAt(context.Background(), predeploys.L1BlockAddr, genesis.ImplementationSlot, latestBlock.Number())
+	require.NoError(t, err)
+	assert.Equal(t, expectedL1BlockAddress, common.BytesToAddress(updatedL1BlockAddress))
+	assert.NotEqualf(t, initialL1BlockAddress, updatedL1BlockAddress, "L1Block Proxy address should have changed")
+
+	_, err = gasPriceOracle.Scalar(nil)
+	require.ErrorContains(t, err, "scalar() is deprecated")
+
+	cost, err := gasPriceOracle.GetL1Fee(nil, []byte{0, 1, 2, 3, 4})
+	require.NoError(t, err)
+	require.Equal(t, cost.Uint64(), uint64(0), "expecting zero scalars within activation block")
```

Comment:
check also that gasPriceOracle.isEcotone() is true?

Reply:
:+1: it's being checked right below